### PR TITLE
Add body item convert to fix datetime compare issue

### PR DESCRIPTION
**Pull request by pshao25**
# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/arrays/item-types/mockapi.ts [hunk 1]
```diff
@@ -50,7 +65,9 @@ const Float32ValueMock = createModelMockApis("float32", [42.42]);
 Scenarios.Arrays_ItemTypes_Float32Value_get = passOnSuccess(Float32ValueMock.get);
 Scenarios.Arrays_ItemTypes_Float32Value_put = passOnSuccess(Float32ValueMock.put);
 
-const DatetimeValueMock = createModelMockApis("datetime", ["2022-08-26T18:38:00Z"]);
+const DatetimeValueMock = createModelMockApis("datetime", ["2022-08-26T18:38:00Z"], (datetime) =>
+  new Date(datetime).toISOString(),
```
Comment (changlong-liu) [inline packages/cadl-ranch-specs/http/arrays/item-types/mockapi.ts:69]: Hi @pshao25 , will this change payload sending in wire? what are the changes?

Comment (pshao25) [reply packages/cadl-ranch-specs/http/arrays/item-types/mockapi.ts:69]: No, it is a similar fix to https://github.com/Azure/cadl-ranch/blob/7d51edb53af6363b305562d02b728291abfaa46e/packages/cadl-ranch-specs/http/models/property-optional/mockapi.ts#L36-L39

Comment (iscai-msft) [reply packages/cadl-ranch-specs/http/arrays/item-types/mockapi.ts:69]: if it's a similar fix can we create a shared helper between these? it seems like we'll keep running into this issue

**Review by iscai-msft — APPROVED** [review]
looks great!

**Pull request merged**